srml/bridge/src/ancestry_proof.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+
+//! A way to check whether or not two headers are related to eachother.
+
+use crate::error::Error;
+
+// Question: Since this is from std, am I allowed to use it?
+use std::collections::HashMap;
+use sr_primitives::traits::Header;
+
+/// A struct used to check whether or not two headers
+/// are related to one another.
+pub struct AncestryProofChecker<H: Header> {
```

Comment:
Does this need to be a struct? I feel like a simple `verify_ancestry` function is fine.